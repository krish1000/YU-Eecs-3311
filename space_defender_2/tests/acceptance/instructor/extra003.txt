--Test debug mode toggle.
play(10,30,2,2,2,2,2)
setup_next(5)
move(D,9)
toggle_debug_mode
pass
toggle_debug_mode
toggle_debug_mode
toggle_debug_mode
move(D,16)
toggle_debug_mode
pass
toggle_debug_mode
pass
