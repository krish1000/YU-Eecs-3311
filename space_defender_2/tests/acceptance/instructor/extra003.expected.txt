  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->play(10,30,2,2,2,2,2)
  state:weapon setup, normal, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_next(5)
  state:in game(0.0), normal, ok
  Starfighter:
    [0,S]->health:70/70, energy:70/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[E,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    B _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    C _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    D _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    E S  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    F _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    G _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    H _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    I _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    J _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
->move(D,9)
  state:in game(1.0), normal, ok
  Starfighter:
    [0,S]->health:70/70, energy:43/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[D,9]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?
    D _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    G _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    I _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
    J ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?  ?
->toggle_debug_mode
  state:in game(1.1), debug, ok
  In debug mode.
->pass
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:49/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[D,9]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:110/110, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,28]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,27]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [D,9], doubling regen rate.
  Enemy Action:
    A Grunt(id:1) gains 10 total health.
    A Grunt(id:1) moves: [A,30] -> [A,28]
      A enemy projectile(id:-1) spawns at location [A,27].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  G  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->toggle_debug_mode
  state:in game(2.1), normal, ok
  Not in debug mode.
->toggle_debug_mode
  state:in game(2.2), debug, ok
  In debug mode.
->toggle_debug_mode
  state:in game(2.3), normal, ok
  Not in debug mode.
->move(D,16)
  state:in game(3.0), normal, ok
  Starfighter:
    [0,S]->health:70/70, energy:31/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[D,16]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  _  <  G  ?  ?  ?  ?
    B ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?
    C ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?
    D ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  ?
    E ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?
    F ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?
    G ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?
    H ?  ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?
    I ?  ?  ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?
    J ?  ?  ?  ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?
->toggle_debug_mode
  state:in game(3.1), debug, ok
  In debug mode.
->pass
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:37/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[D,16]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:120/120, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,24]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,19]
    [-2,<]->damage:15, move:4, location:[A,21]
    [-3,<]->damage:15, move:4, location:[A,23]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,23] -> [A,19]
    A enemy projectile(id:-2) moves: [A,25] -> [A,21]
  Starfighter Action:
    The Starfighter(id:0) passes at location [D,16], doubling regen rate.
  Enemy Action:
    A Grunt(id:1) gains 10 total health.
    A Grunt(id:1) moves: [A,26] -> [A,24]
      A enemy projectile(id:-3) spawns at location [A,23].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  _  <  _  <  G  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->toggle_debug_mode
  state:in game(4.1), normal, ok
  Not in debug mode.
->pass
  state:in game(5.0), normal, ok
  Starfighter:
    [0,S]->health:70/70, energy:43/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[D,16]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  <  _  <  _  <  _  <  G  _  _  _  _  ?  ?  ?  ?
    B ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?
    C ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?
    D ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  ?
    E ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?
    F ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?
    G ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?
    H ?  ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?
    I ?  ?  ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?
    J ?  ?  ?  ?  ?  ?  ?  ?  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  ?  ?  ?  ?  ?  ?  ?
